src/bytes_mut.rs

```diff
@@ -823,12 +761,11 @@ impl BytesMut {
             return Ok(());
         }
 
+        // Note that the following comparison will also yield true for 2 empty
+        // BytesMut. This will however be fine - it will simply combine them
+        // into a single empty BytesMut.
         let ptr = unsafe { self.ptr.as_ptr().offset(self.len as isize) };
```

Comment:
From https://doc.rust-lang.org/std/primitive.pointer.html#safety-2:

> If any of the following conditions are violated, the result is Undefined Behavior:
> ...
> * The computed offset, in bytes, cannot overflow an isize.
> ...
> The compiler and standard library generally tries to ensure allocations never reach a size where an offset is a concern. For instance, Vec and Box ensure they never allocate more than isize::MAX bytes, so vec.as_ptr().add(vec.len()) is always safe.
>
> Most platforms fundamentally can't even construct such an allocation. For instance, no known 64-bit platform can ever serve a request for 2^63 bytes due to page-table limitations or splitting the address space. However, some 32-bit and 16-bit platforms may successfully serve a request for more than isize::MAX bytes with things like Physical Address Extension. As such, memory acquired directly from allocators or memory mapped files may be too large to handle with this function.

Do we put a cap on `len` anywhere to make sure this holds?

Reply:
I don't think we currently do. But we can limit the maximum capacity to `isize::MAX` in `reserve_inner`